File: test-lib/src/main/java/com/flyjingfish/test_lib/mycut/MyAnnoCut.java
package com.flyjingfish.test_lib.mycut;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.flyjingfish.light_aop_annotation.BasePointCut;
import com.flyjingfish.light_aop_annotation.ProceedJoinPoint;
import com.flyjingfish.test_lib.annotation.MyAnno;

public class MyAnnoCut implements BasePointCut<MyAnno> {
    @Nullable
    @Override
    public Object invoke(@NonNull ProceedJoinPoint joinPoint, @NonNull MyAnno anno) {
        return null;
    }
}

File: test-lib/src/main/java/com/flyjingfish/test_lib/mycut/MyAnnoCut.kt
package com.flyjingfish.test_lib.mycut

import android.util.Log
import com.flyjingfish.light_aop_annotation.BasePointCut
import com.flyjingfish.test_lib.annotations.MyAnno
import org.aspectj.lang.ProceedingJoinPoint

class MyAnnoCut : BasePointCut<com.flyjingfish.test_lib.annotations.MyAnno> {
    override fun invoke(
        joinPoint: ProceedingJoinPoint,
        anno: com.flyjingfish.test_lib.annotations.MyAnno
    ): Any? {
        Log.e("MyAnnoCut","===invoke===")
        joinPoint.proceed()
        return null
    }
}